service/jobs/ext_api_test.go

```diff
@@ -0,0 +1,299 @@
+						{
+							RunId: 1234,
+						},
+					},
+					Tasks: []RunTask{
+						{
+							RunId: 999,
+						},
+					},
+					NextPageToken: "token1",
+				},
+			},
+			{
+				Method:       "GET",
+				ReuseRequest: true,
+				Resource:     "/api/2.2/jobs/runs/get?page_token=token1&run_id=4444",
+				Response: Run{
+					Iterations: []RunTask{
+						{
+							RunId: 222,
+						},
+						{
+							RunId: 333,
+						},
+					},
+					Tasks: []RunTask{
+						{
+							RunId: 999,
+						},
+					},
+					PrevPageToken: "token1-reverse",
+				},
+			},
+		},
+		//API 2.1 stubs
+		[]qa.HTTPFixture{
+			{
+				Method:       "GET",
+				ReuseRequest: true,
+				Resource:     "/api/2.1/jobs/runs/get?run_id=514594995218126",
+				Response: Run{
+					Iterations: []RunTask{},
+					Tasks: []RunTask{
+						{
+							RunId:   123,
+							TaskKey: "task1",
+						},
+						{
+							RunId:   1234,
+							TaskKey: "task2",
+						},
+					},
+				},
+			},
+			{
+				Method:       "GET",
+				ReuseRequest: true,
+				Resource:     "/api/2.1/jobs/runs/get?run_id=111222333",
+				Response: Run{
+					Iterations: []RunTask{},
+					Tasks: []RunTask{
+						{
+							RunId: 123,
+						},
+						{
+							RunId: 1234,
+						},
+						{
+							RunId: 222,
+						},
+						{
+							RunId: 333,
+						},
+					},
+					JobClusters: []JobCluster{
+						{
+							JobClusterKey: "cluster1",
+						},
+						{
+							JobClusterKey: "cluster2",
+						},
+					},
+				},
+			},
+			{
+				Method:       "GET",
+				ReuseRequest: true,
+				Resource:     "/api/2.1/jobs/runs/get?run_id=4444",
+				Response: Run{
+					Iterations: []RunTask{
+						{
+							RunId: 123,
+						},
+						{
+							RunId: 1234,
+						},
+						{
+							RunId: 222,
+						},
+						{
+							RunId: 333,
+						},
+					},
+					Tasks: []RunTask{
+						{
+							RunId: 999,
+						},
+					},
+				},
+			},
+		}...,
+	)
+
+}
+
+func TestGetRun(t *testing.T) {
+	ctx := context.Background()
+
+	t.Run("successful run with no pagination", func(t *testing.T) {
+		client, server := commonFixtureWithStatusResponse().Client(t)
+		defer server.Close()
+
+		mockJobsImpl := &jobsImpl{
+			client: client,
+		}
+		api := &JobsAPI{jobsImpl: *mockJobsImpl}
+
+		request := GetRunRequest{RunId: 514594995218126}
+		run, err := api.GetRun(ctx, request)
+
+		assert.NoError(t, err)
+		assert.EqualValues(t, 123, run.Tasks[0].RunId)
+		assert.EqualValues(t, 1234, run.Tasks[1].RunId)
+	})
+
+	t.Run("successful run with two tasks pages", func(t *testing.T) {
+		client, server := commonFixtureWithStatusResponse().Client(t)
+		defer server.Close()
+
+		mockJobsImpl := &jobsImpl{
+			client: client,
+		}
+		api := &JobsAPI{jobsImpl: *mockJobsImpl}
+
+		request := GetRunRequest{RunId: 111222333}
+		run, err := api.GetRun(ctx, request)
+
+		assert.NoError(t, err)
+		assert.Equal(t, 4, len(run.Tasks))
+		assert.Empty(t, run.Iterations)
+		assert.Empty(t, run.NextPageToken)
+		assert.Empty(t, run.PrevPageToken)
```

Comment:
Not now, but longer-term, we may want to hide these fields from the SDK in general. cc @hectorcast-db @renaudhartert-db